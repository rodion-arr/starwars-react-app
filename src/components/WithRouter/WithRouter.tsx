import React from "react";
import {useRoutes} from "react-router-dom";
import {RouteObject} from "react-router";

interface Props {
  routes: RouteObject[]
}

export const WithRouter = ({ routes }: Props): React.ReactElement | null => {
  return useRoutes(routes);
};
