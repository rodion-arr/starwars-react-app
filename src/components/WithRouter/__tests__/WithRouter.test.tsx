import {render} from "@testing-library/react";
import {WithRouter} from "../WithRouter";
import {BrowserRouter as Router} from "react-router-dom";

describe('WithRouter', () => {
  it('should render routes', () => {
    const TestComponent = (): JSX.Element => <>rendered</>
    const routes = [{
      element: <TestComponent />,
      path: '*',
    }]
    const {container} = render(<Router><WithRouter routes={routes} /></Router>);

    expect(container).toMatchSnapshot();
  });
});
