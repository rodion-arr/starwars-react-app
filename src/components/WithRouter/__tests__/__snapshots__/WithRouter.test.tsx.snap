// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithRouter should render routes 1`] = `
<div>
  rendered
</div>
`;
